File: frontend/src/types/notification.ts
export interface Notification {
  notification_id: string;
  user_id: string;
  itinerary_id?: string | null;
  activity_id?: string | null;
  title: string;
  body: string;
  severity: "info" | "warning" | "critical";
  created_at: string;
  read: boolean;
}

File: backend/app/models/notification.py
from pydantic import BaseModel
from typing import Optional, Literal


class Notification(BaseModel):
    notification_id: str
    user_id: str
    itinerary_id: Optional[str] = None
    activity_id: Optional[str] = None
    title: str
    body: str
    severity: Literal["info", "warning", "critical"] = "info"
    created_at: str
    read: bool = False
